donut/modules/uploads/routes.py

```diff
@@ -0,0 +1,69 @@
+import flask
+import json
+import os
+from werkzeug import secure_filename
+
+from donut.modules.uploads import blueprint, helpers
+from donut.resources import Permissions
+from donut.auth_utils import check_permission
+
+
+@blueprint.route('/lib/<path:url>')
+def display(url):
+    '''
+        Displays the webpages that have been created by users.
+    '''
+    page = helpers.read_page(url.replace(' ', '_'))
+    return flask.render_template(
+        'page.html', page=page, title=url.replace('_', ' '), permissions=True)
+    # permission=check_permission(Permissions.ADMIN))
+
+
+@blueprint.route('/uploads', methods=['GET', 'POST'])
+def uploads():
+    '''
+    Serves the webpage that allows a user to upload a file.
+    '''
+    if flask.request.method == 'POST':
+        if 'file' not in flask.request.files:
+            flask.flash('No file part')
+            return flask.render_template('uploads.html')
+        file = flask.request.files['file']
```

Comment:
Can we check that the user is logged in and the file is of a reasonable size (e.g. < 10 MB)? Otherwise it would be very easy to fill up the EC2 instance's hard drive.